atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/ConcludedTransactionsUpdaterTask.java

```diff
@@ -0,0 +1,116 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.sweep;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Range;
+import com.palantir.atlasdb.sweep.queue.ShardAndStrategy;
+import com.palantir.atlasdb.sweep.queue.ShardProgress;
+import com.palantir.atlasdb.sweep.queue.SweepQueueUtils;
+import com.palantir.atlasdb.table.description.SweepStrategy.SweeperStrategy;
+import com.palantir.atlasdb.transaction.impl.TransactionConstants;
+import com.palantir.atlasdb.transaction.knowledge.CoordinationAwareKnownConcludedTransactionsStore;
+import java.util.Comparator;
+import java.util.Set;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public final class ConcludedTransactionsUpdaterTask implements AutoCloseable {
+    public static final int CONCLUDED_TRANSACTIONS_UPDATE_INITIAL_DELAY_MILLIS = 1000;
+    public static final int CONCLUDED_TRANSACTIONS_UPDATE_TASK_DELAY_MILLIS = 5000;
+
+    // This will to be ShardProgress#getNumberOfShards(). Not using the persisted value here can lead to SEVERE DATA
+    // CORRUPTION.
+    private final Supplier<Integer> persistedNumShardsSupplier;
+
+    private final ShardProgress progress;
+    private final CoordinationAwareKnownConcludedTransactionsStore concludedTransactionsStore;
+    private final ScheduledExecutorService executor;
+
+    private ConcludedTransactionsUpdaterTask(
+            Supplier<Integer> shardsSupplier,
+            CoordinationAwareKnownConcludedTransactionsStore concludedTransactionsStore,
+            ShardProgress progress,
+            ScheduledExecutorService executorService) {
+        this.persistedNumShardsSupplier = shardsSupplier;
+        this.concludedTransactionsStore = concludedTransactionsStore;
+        this.progress = progress;
+        this.executor = executorService;
+    }
+
+    public static ConcludedTransactionsUpdaterTask create(
+            Supplier<Integer> numShards,
+            CoordinationAwareKnownConcludedTransactionsStore concludedTransactionsStore,
+            ShardProgress progress,
+            ScheduledExecutorService executor) {
+        ConcludedTransactionsUpdaterTask task =
+                new ConcludedTransactionsUpdaterTask(numShards, concludedTransactionsStore, progress, executor);
+        task.schedule();
+        return task;
+    }
+
+    private void schedule() {
```

Comment:
This is reasonable, we can alternatively consider just running this task on the sweep thread at the end

Reply:
yeah i think that _should_ be fine but could be more coupling than I would like. I will leave a comment for now and we can consider this while wiring - w/e makes most sense.